pos_voucher/models/pos_voucher.py

```diff
@@ -0,0 +1,151 @@
+
+from odoo import api, fields, models, _, SUPERUSER_ID
+from odoo.exceptions import Warning
+
+
+class VoucherType(models.Model):
+    _name = 'voucher.type'
+    _description = 'POS Voucher Type'
+
+    name = fields.Char('Type', required=True)
+    journal_id = fields.Many2one('account.journal', 'Journal', required=True)
+    sequence_id = fields.Many2one('ir.sequence', required=True)
+
+
+class POSVoucher(models.Model):
+    _name = 'pos.voucher'
+    _inherit = ['mail.thread']
+    _rec_name = 'code'
+    _description = 'POS Voucher'
+
+    code = fields.Char('Code', required=True, default='/', copy=False)
+    start_date = fields.Datetime(required=True)
+    end_date = fields.Datetime()
+    type_id = fields.Many2one('voucher.type', 'Type', required=True)
+    customer_id = fields.Many2one('res.partner', 'Customer', required=True)
+    consume_date = fields.Datetime('Consumed on',
+                                   compute='_compute_date_amount')
+    discount_type = fields.Selection([('fixed', 'Fixed')], default='fixed')
+    total_amount = fields.Float(digits=(16, 2))
+    pending_amount = fields.Float(compute='_compute_date_amount',
+                                  store=True, digits=(16, 2))
+    voucher_history_ids = fields.One2many('pos.voucher.history',
+                                          'pos_voucher_id',
+                                          'Voucher History')
+    state = fields.Selection([('draft', 'Draft'),
+                              ('validated', 'Validated'),
+                              ('partially_consumed', 'Partially Consumed'),
+                              ('consumed', 'Consumed'),
+                              ('expired', 'Expired'),
+                              ('cancelled', 'Cancelled')], default='draft',
+                             required=True, copy=False)
+    _sql_constraints = [
+        ('unique_code', 'unique (code)',
+         'The Code of the Voucher must be unique!'),
+    ]
+
+    @api.depends('total_amount', 'voucher_history_ids',
+                 'voucher_history_ids.amount',
+                 'voucher_history_ids.consumed_date')
+    def _compute_date_amount(self):
```

Comment:
The performance of this function could be improved with a read_group call.
It's not a priority, but this comment would be ok ^^

```suggestion
    def _compute_date_amount(self):
        # TODO: Use read_group to improve performance
```